etc/init.d/zfs-load-key.in

```diff
@@ -0,0 +1,131 @@
+#!@DEFAULT_INIT_SHELL@
+#
+# zfs-load-key  This script will load/unload the zfs filesystems keys.
+#
+# chkconfig:    2345 06 99
+# description:  This script will load or unload the zfs filesystems keys during
+#               system boot/shutdown. Only filesystems with key path set
+#               in keylocation property. See the zfs(8) man page for details.
+# probe: true
+#
+### BEGIN INIT INFO
+# Provides:          zfs-load-key
+# Required-Start:    $local_fs zfs-import
+# Required-Stop:     $local_fs zfs-import
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# X-Start-Before:    zfs-mount
+# X-Stop-After:      zfs-zed
+# Short-Description: Load ZFS keys for filesystems and volumes
+# Description: Run the `zfs load-key` or `zfs unload-key` commands.
+### END INIT INFO
+#
+# Released under the 2-clause BSD license.
+#
+# This script is based on debian/zfsutils.zfs.init from the
+# Debian GNU/kFreeBSD zfsutils 8.1-3 package, written by Aurelien Jarno.
+
+# Source the common init script
+. @sysconfdir@/zfs/zfs-functions
+
+# ----------------------------------------------------
+
+do_depend()
+{
+	# bootmisc will log to /var which may be a different zfs than root.
+	before bootmisc logger zfs-mount
+
+	after zfs-import sysfs
+	keyword -lxc -openvz -prefix -vserver
+}
+
+# Load keys for all datasets/filesystems
+do_load_keys()
+{
+	zfs_log_begin_msg "Load ZFS filesystem(s) keys"
+
+	"$ZFS" list -Ho name,encryptionroot,keystatus,keylocation |
+		while read -r name encryptionroot keystatus keylocation; do
+			if [ "$encryptionroot" != "-" ] &&
+				[ "$name" = "$encryptionroot" ] &&
+				[ "$keystatus" = "unavailable" ] &&
+				[ "$keylocation" != "prompt" ] &&
+				[ "$keylocation" != "none" ]
+			then
+			zfs_action "Load key for $encryptionroot" \
+				"$ZFS" load-key "$encryptionroot"
```

Comment:
tab for continuation (should be 4 spaces)

Reply:
Mixing spaces an tabs is normal in bash? Wouldn't this break if someone has configured another tab representation? (eg 2spaces).

Reply:
bash seems to use a mix of 2-space and emacs indent, but I don't really see how that's relevant

Reply:
I will do a cleanup.